PowerPointLabs/PowerPointLabs/AudioMisc/Audio.cs

```diff
@@ -41,6 +45,27 @@ public Audio(string name, string saveName, int matchScriptID)
             Type = AudioHelper.GetAudioType(saveName);
         }
 
+        public Audio(Shape shape, string saveName)
+        {
+            // detect audio type from shape tag
+            AudioType audioType;
+            if (!Enum.TryParse<AudioType>(shape.Tags[AudioTypeTagName], out audioType))

```

Comment:
quick question,
for a new audio shape, it won't have the audio type.
so it's still using the sampling rate to distinguish `recorded` & `generated`.
does it fix the issue?